### ISSUE TITLE
[Accessibility]The error message can not be read by screen reader tool

### ISSUE THREAD

**Issue by lukewang2018** [carbon-design-system/carbon#3645]
##[Accessibility]The error message can not be read by screen reader tool

## Environment
> Operating system
Any
> Browser
Any

## Detailed description

> What version of the Carbon Design System are you using?
"carbon-components-react": "^6.115.5",

> What did you expect to happen?
It should have have aria fields as below.  refer to TextInput.
_data-invalid="true" 
aria-invalid="true" 
aria-describedby="name-input-error-msg"_

> What happened instead?
Not compliance.

> What WCAG 2.1 checkpoint does the issue violate?
Accessibility checkpoint: IBM 3.3.1 Error Identification
( https://www.ibm.com/able/guidelines/ci162/error_id.html )

## Steps to reproduce the issue

1. Use the component NumberInput
2. Input something wrong in the field
3. check the source elements
4. There should be an error message node and it's id is referenced by aria-describedby. 

 

- Screenshots or code
<img width="611" alt="Screen Shot 2019-08-05 at 11 07 12 AM" src="https://user-images.githubusercontent.com/40848250/62436127-392dc880-b771-11e9-870c-cbeeeddb7035.png">
